lib/Service/StackService.php

```diff
@@ -362,4 +368,93 @@ public function reorder($id, $order) {
 
 		return $result;
 	}
+
+	/**
+	 * @param $id
+	 * @param $boardId
+	 * @return Stack
+	 * @throws StatusException
+	 * @throws BadRequestException
+	 */
+	public function clone($id, $boardId) {
+		if (is_numeric($id) === false) {
+			throw new BadRequestException('stack id must be a number');
+		}
+
+		$this->permissionService->checkPermission(null, $boardId, Acl::PERMISSION_MANAGE);
+		if ($this->boardService->isArchived(null, $boardId)) {
+			throw new StatusException('Operation not allowed. This board is archived.');
+		}
+
+		$stack = $this->stackMapper->find($id);
+
+		$newStack = new Stack();
+		$newStack->setTitle($stack->getTitle() . ' (' . $this->l10n->t('copy') . ')');
+		$newStack->setBoardId($boardId);
+		$newStack->setOrder($stack->getOrder() +1);
+		$newStack = $this->stackMapper->insert($newStack);
+
+		$cards = $this->cardMapper->findAll($id);
+		foreach ($cards as $card) {
+			$newCard = new Card();
```

Comment:
I think it should also work to just operate on $card directly so you just overwrite the properties that need to be changed and then passed to insert afterwards.